### PR TITLE
Adding missing readline createInterface options

### PR DESCRIPTION
**Pull request by ghost**
prompt?: string;
crlfDelay?: number;
removeHistoryDuplicates?: boolean;

Please fill in this template.

- [Adding missing readline createInterface options ] Use a meaningful title for the pull request. Include the name of the package modified.
- [Works ] Test the change in your own code. (Compile and run.)
- [ Yes] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [Yes ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [Complete ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ Complete ] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://nodejs.org/api/readline.html>>
- [ Unsure about this ] Increase the version number in the header if appropriate.
- [ Minor Changes ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by McClellandLaboratories** [comment]
Looks like this line 1591:

export function createInterface(input: NodeJS.ReadableStream, output?: NodeJS.WritableStream, completer?: Completer | AsyncCompleter, terminal?: boolean): ReadLine;

should be updated to:

export function createInterface(input: NodeJS.ReadableStream, output?: NodeJS.WritableStream, completer?: Completer | AsyncCompleter, terminal?: boolean, historySize?: number, prompt?: string, crlfDelay?: number, removeHistoryDuplicates?: boolean): ReadLine;

File: types/node/v7/index.d.ts [hunk 1]
```diff
@@ -1583,9 +1583,12 @@ declare module "readline" {
         completer?: Completer | AsyncCompleter;
         terminal?: boolean;
         historySize?: number;
+        prompt?: string;
+        crlfDelay?: number;
+        removeHistoryDuplicates?: boolean;
     }
 
-    export function createInterface(input: NodeJS.ReadableStream, output?: NodeJS.WritableStream, completer?: Completer | AsyncCompleter, terminal?: boolean): ReadLine;
+    export function createInterface(input: NodeJS.ReadableStream, output?: NodeJS.WritableStream, completer?: Completer | AsyncCompleter, terminal?: boolean, historySize?: number, prompt?: string, crlfDelay?: number, removeHistoryDuplicates?: boolean): ReadLine;
```
Comment (rbuckton) [inline types/node/v7/index.d.ts]: While the changes to `ReadLineOptions` look correct, there is no overload for `createInterface` that takes more than 4 parameters.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/node/v7/index.d.ts (hunk 1 above)
Comment (McClellandLaboratories) [inline types/node/v7/index.d.ts]: Hmm, I have been able to successfully use createInterface with 8 parameters in the past. However, I am OK with this edit, its easy enough to use the ReadLineOptions format. This is my first time contributing, do I need to do anything further?

Comment (rbuckton) [inline types/node/v7/index.d.ts]: Only the first four parameters are actually used, as far as I can tell. 

Before I can accept this PR you would need to commit the change mentioned above and push your changes. Once that is done I can merge.

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**